pkg/registry/storage/volumeattachment/strategy.go

```diff
@@ -53,13 +55,20 @@ func (volumeAttachmentStrategy) PrepareForCreate(ctx context.Context, obj runtim
 		groupVersion = schema.GroupVersion{Group: requestInfo.APIGroup, Version: requestInfo.APIVersion}
 	}
 
+	volumeAttachment := obj.(*storage.VolumeAttachment)
```

Comment:
shouldn't we check that the cast was successful ?

Reply:
no, this is well tested and is a programmer error if any other type comes through here

Reply:
What do you think of the following ?
```
diff --git a/pkg/registry/storage/volumeattachment/strategy.go b/pkg/registry/storage/volumeattachment/strategy.go
index 183fe02d95..b3468d9370 100644
--- a/pkg/registry/storage/volumeattachment/strategy.go
+++ b/pkg/registry/storage/volumeattachment/strategy.go
@@ -48,20 +48,17 @@ func (volumeAttachmentStrategy) NamespaceScoped() bool {
 }

 // ResetBeforeCreate clears the Status field which is not allowed to be set by end users on creation.
-func (volumeAttachmentStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
+func (volumeAttachmentStrategy) PrepareForCreate(ctx context.Context, volumeAttachment *storage.VolumeAttachment) {
        var groupVersion schema.GroupVersion

        if requestInfo, found := genericapirequest.RequestInfoFrom(ctx); found {
                groupVersion = schema.GroupVersion{Group: requestInfo.APIGroup, Version: requestInfo.APIVersion}
        }

-       volumeAttachment := obj.(*storage.VolumeAttachment)
-
        switch groupVersion {
        case storageapiv1beta1.SchemeGroupVersion:
                // allow modification of status for v1beta1
        default:
-               volumeAttachment := obj.(*storage.VolumeAttachment)
                volumeAttachment.Status = storage.VolumeAttachmentStatus{}
        }
```
This would make the parameter type more visible.

thanks

Reply:
No, PrepareForCreate is an interface all strategies implement

Reply:
thanks for the explanation.